js/client-teambuilder.js

```diff
@@ -1699,19 +1699,33 @@
 
 			if (template.speciesid === 'meowstic') {
 				smogdexid = 'meowstic-m';
-			} else if (smogdexid === 'rotom' || smogdexid === 'deoxys' || smogdexid === 'kyurem' || smogdexid === 'giratina' || smogdexid === 'shaymin' || smogdexid === 'tornadus' || smogdexid === 'thundurus' || smogdexid === 'landorus' || smogdexid === 'pumpkaboo' || smogdexid === 'gourgeist' || smogdexid === 'arceus' || smogdexid === 'meowstic' || smogdexid === 'hoopa') {
-				if (template.forme) smogdexid += '-' + toId(template.forme);
+			} else if (template.forme) {
+				if (template.speciesid === 'vivillonfancy' || template.speciesid === 'vivillonpokeball') {
+					smogdexid = 'vivillon';
+				} else if (template.speciesid === 'keldeoresolute') {
+					smogdexid = 'keldeo';
+				} else if (template.speciesid === 'basculinbluestriped') {
+					smogdexid = 'basculin';
+				} else if (template.speciesid === 'greninjaash') {
+					smogdexid = 'greninja';
+				} else if (template.speciesid === 'pikachubelle' || template.speciesid === 'pikachuphd' || template.speciesid === 'pikachupopstar' || template.speciesid === 'pikachurockstar' || template.speciesid === 'pikachulibre' || template.speciesid === 'pikachucosplay') {
+					smogdexid = 'pikachu';
+				} else if (template.speciesid === 'castformsunny' || template.speciesid === 'castformsnowy' || template.speciesid === 'castformrainy') {
+					smogdexid = 'castform';
```

Comment:
This is a lot of code duplication. You'd probably want something like a switch at the very least. And you should probably switch on `baseSpecies` so you don't have to list every single forme of the formes you do need to list.